### PR TITLE
Remove git submodules, run remotely hosted replica_server

### PR DESCRIPTION
**Pull request by sgwilym**
- Removed git submodules for earthstar, replica_server, and earthstar_streaming_rpc
- Made it so that this file runs a remotely hosted version of replica_server instead
- Changed how allow_list.json is loaded for this new setup
- Make Docker ignore .git (having all those submodules in history made deploying this thing looooong)

### CODE REVIEW

**Comment by sandreae** [comment]
Hey, thanks for this, great that the modules are now pulled down from the remotes.

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
 EXPOSE 8080
+EXPOSE 443
 
 WORKDIR /app
 
 USER deno
 
-ADD . .
+COPY allow_list.json ./allow_list.json
```
Comment (sandreae) [inline Dockerfile]: As I understand it, it's best practice to not copy sensitive data across in the build process, better to pass them in at runtime. I tried to do this in the previous version but actually see that it wasn't doing what I wanted then anyway. I suggest removing this line and then moving `allow_list.json` across in the docker run command.

**Review by sandreae — CHANGES_REQUESTED** [review]
Thanks for following up on this repo so quickly!

I had one comment about how we are moving the allowed list over (see below). Something similar applies for the generated `data/` folder. If we want this to persist then a volume for this should be mounted too. Then we get permission issues if the host and container users aren't matching (sigh... i never said I liked containers), so we need to set permissions and mount a volume for this at runtime too. 

The dockerfile would become like this:

```dockerfile
FROM denoland/deno:1.19.0

EXPOSE 8080

WORKDIR /app

RUN mkdir /app/data/ \
    && chown deno:deno /app/data/

VOLUME [ "/app/data" ]

USER deno

RUN deno cache https://deno.land/x/earthstar_replica_server@v1.0.1/server.ts
CMD ["run", "--allow-all", "--no-check", "https://deno.land/x/earthstar_replica_server@v1.0.1/server.ts"]
```  

and the run command like so:

```bash
docker build -t earthstar-replica-server . && docker run -v $(pwd)/allow_list.json:/app/allow_list.json -v data:/app/data -it --init -p 8080:8080 earthstar-replica-server
```

File: Dockerfile (hunk 1 above)
Comment (sgwilym) [reply Dockerfile]: Hey, I've made these changes and added a makefile to make running the command a bit easier (i.e. `make docker`). Shame about the add / copy thing being insecure, I think it means I can't reuse this dockerfile for deployment with Fly.io (as you don't get to call `docker run` yourself in that case).

I also updated replica-server to take an optional flag for the path of the allow list (which I use here).

Comment (sandreae) [reply Dockerfile]: Maybe there would be a way to setup a volume for `allow_list.json` on fly.io?

Another approach could be to pass it in as an environment variable.

I don't really know _how_ insecure adding the file in the build is, someone would have to get into your server first to be able to check out your build images.... which is already a bad security breach...

**Comment by sandreae** [comment]
Thanks for this, I'm gunna merge. 

I think we can find a solution for deploying on fly.io as well, but we can do that in a separate PR.

**Pull request merged**